main.go

```diff
@@ -1,639 +1,60 @@
-
-	err := resetLaunchServices()
-	if err != nil {
-		log.Warnf("Failed to apply simulator boot workaround, error: %s", err)
-	}
-
-	// Boot simulator
-	if cfg.SimulatorDebug != never {
-		log.Infof("Enabling Simulator verbose log for better diagnostics")
-		// Boot the simulator now, so verbose logging can be enabled and it is kept booted after running tests,
-		// this helps to collect more detailed debug info
-		if err := simulatorBoot(cfg.SimulatorID); err != nil {
-			return Result{}, fmt.Errorf("%v", err)
-		}
-		if err := simulatorEnableVerboseLog(cfg.SimulatorID); err != nil {
-			return Result{}, fmt.Errorf("%v", err)
-		}
-
-		fmt.Println()
-	}
-
-	if !cfg.IsSimulatorBooted && !cfg.HeadlessMode {
-		log.Infof("Booting simulator (%s)...", cfg.SimulatorID)
-
-		if err := simulator.BootSimulator(cfg.SimulatorID, cfg.XcodeMajorVersion); err != nil {
-			if err := cmd.ExportEnvironmentWithEnvman("BITRISE_XCODE_TEST_RESULT", "failed"); err != nil {
-				log.Warnf("Failed to export: BITRISE_XCODE_TEST_RESULT, error: %s", err)
-			}
-			return Result{}, fmt.Errorf("failed to boot simulator, error: %s", err)
-		}
-
-		progress.NewDefaultWrapper("Waiting for simulator boot").WrapAction(func() {
-			time.Sleep(60 * time.Second)
-		})
-
-		fmt.Println()
-	}
-
-	// Run build
-	result := Result{}
-
-	projectFlag := "-project"
-	if filepath.Ext(cfg.ProjectPath) == ".xcworkspace" {
-		projectFlag = "-workspace"
-	}
-
-	buildParams := models.XcodebuildParams{
-		Action:                    projectFlag,
-		ProjectPath:               cfg.ProjectPath,
-		Scheme:                    cfg.Scheme,
-		DeviceDestination:         fmt.Sprintf("id=%s", cfg.SimulatorID),
-		CleanBuild:                cfg.IsCleanBuild,
-		DisableIndexWhileBuilding: cfg.DisableIndexWhileBuilding,
-	}
-
-	if !cfg.IsSingleBuild {
-		buildLog, exitCode, err := runBuild(buildParams, cfg.OutputTool)
-		result.XcodebuildBuildLog = buildLog
-		if err != nil {
-			log.Warnf("xcode build exit code: %d", exitCode)
-			log.Warnf("xcode build log:\n%s", buildLog)
-			log.Errorf("xcode build failed with error: %s", err)
-			return result, err
-		}
-	}
-
-	// Run test
-	tempDir, err := ioutil.TempDir("", "XCUITestOutput")
+func run() int {
+	logger := log.NewLogger()
+	envRepository := env.NewRepository()
+	inputParser := stepconf.NewInputParser(envRepository)
+	xcprettyInstaller := xcpretty.NewInstaller()
+	commandFactory := command.NewFactory(envRepository)
+	pathChecker := pathutil.NewPathChecker()
+	fileRemover := fileremover.NewFileRemover()
+	xcodebuilder := xcodebuild.New(logger, commandFactory, pathChecker, fileRemover)
+	sim := simulator.New()
+	c := cache.NewCache()
+	testAddonExporter := testaddon.NewExporter()
+	testArtifactExporter := testartifact.NewExporter()
+	stepenvRepository := stepenv.NewRepository(envRepository)
+	outputExporter := output.NewExporter(stepenvRepository, logger, testAddonExporter, testArtifactExporter)
+	pathModifier := pathutil.NewPathModifier()
```

Comment:
Let's extract the manual DI to a separate function (e.g. `createStep`).

Reply:
Agree